homeassistant/components/mqtt/client.py

```diff
@@ -543,21 +653,22 @@ def _client_unsubscribe(topic: str) -> int:
             _raise_on_error(result)
             return mid
 
+        if self._is_active_subscription(topic):
```

Comment:
What happens if you had 2 subscriptions, one with QoS 0, one with QoS 1, and you unsubscribe from the QoS 1?
Don't you need to re-subscribe with QoS 0?

ie. how do you handle the situation where the max_qos is lowered due to an unsubscription?

Reply:
Also - possibly add:
```
else:
  if topic in self._max_qos[topic]:
    del self._max_qos[topic]
```

Here.  Because if there are no more subscriptions to a topic, there is no need for an entry in _max_qos

Reply:
That is not needed, if the device uses a lower QoS then thay QoS will be used.